### PR TITLE
fix: cfc sync's removing algorithm

### LINKED ISSUE #506: Dynamic children errors with `Cannot read property 'key' of undefined`
## Description
Having a dynamic children will cause errors in flicking. Is this meant to be supported? As a workaround, you can generate a key based on the children, so it will mount a new component when children change, but this is not ideal.

## Steps to check or reproduce
In this sandbox, the children are randomly generated every second. If you remove the `key` on `<Flicking>`, it will cause an error pretty quickly.
https://codesandbox.io/s/restless-butterfly-ttl8g?file=/src/App.tsx

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#506 

## Details
This fixes #506

### CODE REVIEW

File: src/cfc/sync.ts [hunk 1]
```diff
@@ -10,23 +10,23 @@ export default (flicking: Flicking, diffResult: DiffResult<any>, rendered: any[]
     let startIdx = -1;
     let prevIdx = -1;
 
-    diffResult.removed.forEach((removedIdx, idx) => {
+    [...diffResult.removed].reverse().forEach(removedIdx => {
```
Comment (daybrush) [inline src/cfc/sync.ts:13]: If the order is correct, the index will change every time you delete it, is that ok?

**Review by daybrush — APPROVED** [review]
LGTM

File: src/cfc/sync.ts (hunk 1 above)
Comment (WoodNeck) [reply src/cfc/sync.ts:13]: Good point, I should add a TC for that

**Review by mixed — APPROVED** [review]
LGTM

**Pull request merged**